Sources/Hummingbird/Router/Router+validation.swift

```diff
@@ -0,0 +1,111 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import HTTPTypes
+
+extension Router {
+    /// Route description
+    public struct RouteDescription: CustomStringConvertible {
+        /// Route path
+        public let path: RouterPath
+        /// Route method
+        public let method: HTTPRequest.Method
+
+        public var description: String { "\(method) \(path)" }
+    }
+
+    /// List of routes added to router
+    public var routes: [RouteDescription] {
+        let trieValues = self.trie.root.values()
+        return trieValues.flatMap { endpoint in
+            endpoint.value.methods.keys
+                .sorted { $0.rawValue < $1.rawValue }
+                .map { RouteDescription(path: endpoint.path, method: $0) }
+        }
+    }
+
+    /// Validate router
+    ///
+    /// Verify that routes are not clashing
+    public func validate() throws {
```

Comment:
Would we recommend users to call this themselves? Or do we add an `#if DEBUG` block that calls this somewhere?

Reply:
It would have been good to add to the `Router.buildResponder` function but that isn't a throwing function. I was thinking of adding it to the template.

Reply:
What about adding this?

```swift
#if DEBUG
do {
  try self.validate()
} catch {
  assertionFailure("......")
}
#endif
```

Reply:
I guess I could